src/3p.js

```diff
@@ -109,18 +109,25 @@ function executeAfterWriteScript(win, fn) {
  * @param {string} src
  */
 export function validateSrcPrefix(prefix, src) {
-
+  const validPrefixes = [];
   if (!isArray(prefix)) {
     prefix = [prefix];
   }
-
-  for (const p of prefix) {
-    if (src.indexOf(p) === 0) {
+  const totalPrefixes = prefix.length;
+  if (src !== undefined) {
+    let p = 0;
+    for (p = 0; p <= totalPrefixes; p++) {
+      const protocolIndex = src.toLowerCase().indexOf(prefix[p]);
+      if (protocolIndex == 0) {
+        validPrefixes.push(p.toString());
```

Comment:
Since this function was design to validate 1 or more prefixes ['https:','http:'], in the event we do have matches we want to keep track of it separately, especially if an invalid protocol is presented first, i.e. **http** in the list of prefixes. 

So the flow is structured this way to go after one of the original problems with this function (#2007), that failed it's own test case when accepting 2 or prefixes, causing it to throw errors during the build even when 1 valid prefix was in the list.